src/mc-fle2-encryption-placeholder.c

```diff
@@ -405,26 +437,50 @@ bool mc_FLE2RangeInsertSpec_parse(mc_FLE2RangeInsertSpec_t *out, const bson_iter
 
         IF_FIELD(precision) {
             if (!BSON_ITER_HOLDS_INT32(&iter)) {
-                CLIENT_ERR("invalid FLE2RangeFindSpecEdgesInfo: 'precision' must "
+                CLIENT_ERR("invalid FLE2RangeInsertSpec: 'precision' must "
                            "be an int32");
                 goto fail;
             }
             int32_t val = bson_iter_int32(&iter);
             if (val < 0) {
-                CLIENT_ERR("invalid FLE2RangeFindSpecEdgesInfo: 'precision' must be"
+                CLIENT_ERR("invalid FLE2RangeInsertSpec: 'precision' must be"
                            "non-negative");
                 goto fail;
             }
             out->precision = OPT_U32((uint32_t)val);
         }
         END_IF_FIELD
+
+        IF_FIELD(trimFactor) {
+            if (!BSON_ITER_HOLDS_INT32(&iter)) {
+                CLIENT_ERR("invalid FLE2RangeInsertSpec: 'trimFactor' must "
+                           "be an int32");
+                goto fail;
+            }
+            int32_t val = bson_iter_int32(&iter);
+            if (val < 0) {
+                CLIENT_ERR("invalid FLE2RangeInsertSpec: 'trimFactor' must be"
+                           "non-negative");
+                goto fail;
+            }
+            out->trimFactor = OPT_U32((uint32_t)val);
+        }
+        END_IF_FIELD
     }
 
     CHECK_HAS(v)
     CHECK_HAS(min)
     CHECK_HAS(max)
     // Do not error if precision is not present. Precision optional and only
     // applies to double/decimal128.
+
+    if (out->trimFactor.set && !use_range_v2) {
+        // Once `use_range_v2` is default true, this block may be removed.
+        CLIENT_ERR("invalid FLE2RangeInsertSpec: trimFactor is not supported for QE range v1");
+        return false;
+    }
+    // Do not error if trim factor is not present. It is optional as it requires a feature flag,
```

Comment:
It should be required once you land the server changes. There will never be a case where we don't send it to the server for range even when it is just 0.

Reply:
Changed the logic such that:

- For parse_RangeOpts, trim factor is optional when use_range_v2 is enabled, and causes an error if present otherwise.
- For RangeOpts_to_InsertSpec, when use_range_v2 is enabled, if trim factor is not present, appends trim factor = 0 to the insert spec. If use_range_v2 is disabled, appends nothing.
- For parse_InsertSpec and parse_FindSpecEdgesInfo, trim factor is required when use_range_v2 is enabled, and causes an error if present otherwise.

Also changed the testing around to test edge cases of use_range_v2 + trim factor presence/absence more thoroughly.